### PR TITLE
Bump on requeue

### PR DESCRIPTION
**Pull request by croissanne**
worker/server: update metrics on requeue
    
    When requeuing a job the next worker requesting the job would decrement
    pending counter, but the pending counter only ever got incremented once,
    when the job was first enqueued. Thus make sure to increment the pending
    counter when a job is requeued.

### CODE REVIEW

File: internal/worker/server.go [hunk 1]
```diff
@@ -718,6 +718,14 @@ func (s *Server) RequeueOrFinishJob(token uuid.UUID, maxRetries uint64, result j
 		}
 	}
 
+	if requeued {
+		jobInfo, err := s.jobInfo(jobId, nil)
+		if err != nil {
+			return fmt.Errorf("error requeueing job: %w", err)
```
Comment (schuellerf) [inline internal/worker/server.go:724]: more like `error getting job details for prometheus metrics`
should we skip prometheus metrics here and try to continue, rather than fail?
although this will then probably fail some lines below…

**Review by schuellerf — APPROVED** [review]
LGTM, There is no explicit test for the metrics right?
Not sure if this is doable with a sane effort.
And might be a separate PR?

File: internal/worker/server.go (hunk 1 above)
Comment (croissanne) [reply internal/worker/server.go:724]: I think it's very unlikely that this call will fail if the job exists.

**Comment by croissanne** [comment]
It's definitely possible to add unit tests, I'll do that.

**Comment by schuellerf** [comment]
Sidenote - I think the `running_jobs` metric also has some problems (causing it to go below 0) so we could see this in the tests if you can implement them 🥳

**Review by schuellerf — APPROVED** [review]
LGTM, I'll check additionally if adding more tests to `RunningJobs` metric will bring up problems in some corner cases (as I think there should be some)
